judge/views/submission.py

```diff
@@ -251,7 +251,9 @@ def _get_queryset(self):
                 contest_queryset = Contest.objects.filter(Q(authors=self.request.profile) |
                                                           Q(curators=self.request.profile) |
                                                           Q(scoreboard_visibility=Contest.SCOREBOARD_VISIBLE) |
-                                                          Q(end_time__lt=timezone.now())).distinct()
+                                                          Q(end_time__lt=timezone.now())) \
```

Comment:
Doesn't this prevent contest authors/curators from seeing submissions to their own contests? I think the correct query would be something like:
```suggestion
                                                          Q(end_time__lt=timezone.now(), scoreboard_visibility__in=(Contest.SCOREBOARD_AFTER_PARTICIPATION, Contest.SCOREBOARD_AFTER_CONTEST))) \
```
(and remove the `.exclude()`.

Reply:
Ah, you're right. It might be worth noting that this query doesn't handle AFTER_PARTICIPATION properly.